src/TerminalFighter/TestState/TestState.cpp

```diff
@@ -9,28 +10,40 @@
 #include "Universe/Universe.h"
 #include "CollisionDetector/CollisionDetector.h"
 #include "GraphicsHandler/GraphicsHandler.h"
+#include "Settings/Settings.h"
 
 #include "MainCharacter/MainCharacter.h"
 #include "BasicEnemy/BasicEnemy.h"
 #include "Terminal/Terminal.h"
 #include "TargetingSystem/TargetingSystem.h"
 
-TestState::TestState(SDL_Renderer& renderer)
+TestState::TestState(SDL_Renderer& renderer, const I_Settings& settings)
     : exit_(false)
-    , renderer_(renderer) {
+    , renderer_(renderer)
+    , settings_(settings) {
 }
 
 gamestates::GameStateName TestState::run() {
+    int screen_width = settings_.video_settings()["window"]["width"].as<int>();
+    int screen_height = settings_.video_settings()["window"]["height"].as<int>();
+
+    std::vector<std::string> graphic_paths;
+    auto graphics_node = settings_.asset_paths()["graphics"];
+
+    for (YAML::const_iterator it = graphics_node.begin(); it != graphics_node.end(); ++it) {
```

Comment:
It's funny you used `auto` on line 31 `auto graphics_node`, but not here where it is easy to tell `it` is an iterator ;D. 

I think for readability, line 31 should have it's typed specified, but here is a good case of using `auto` for the iterator. 

https://stackoverflow.com/a/6435066